syncoid

```diff
@@ -837,15 +837,16 @@ sub syncdataset {
 	}
 	if (defined $args{'no-sync-snap'}) {
 		if (defined $args{'create-bookmark'}) {
-			my $ret = createbookmark($sourcehost, $sourcefs, $newsyncsnap, $newsyncsnap);
+			my %existingbookmarks = getbookmarks($sourcehost,$sourcefs,$sourceisroot);
+			my $guid = $snaps{'source'}{$newsyncsnap}{'guid'};
+			my $ret = createbookmark($sourcehost, $sourcefs, $newsyncsnap, $newsyncsnap, $guid, %existingbookmarks);
 			$ret == 0 or do {
```

Comment:
this code should stay as it where before, because doing it this way will degrade performance because each time all the bookmarks needs to be listed.
It's better to handle that in the fallback case (if the bookmark creation fails), there the bookmark listing and the compare if this bookmark already exists should be made.